plugins/modules/ec2_imagebuilder_component.py

```diff
@@ -0,0 +1,399 @@
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    module.exit_json(changed=changed, component=dict(arn=arn))
+
+
+def get_aws_account_id(module):
+    return module.client('sts').get_caller_identity()['Account']
+
+
+def get_component(connection, module, arn):
+    """
+    Get an EC2 Image Builder component by ARN. If not found, return None.
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param arn: ARN of Image Builder component to get
+    :return: boto3 Image Builder component dict or None if not found
+    """
+
+    try:
+        return connection.get_component(componentBuildVersionArn=arn)['component']
+    except (BotoCoreError, ClientError) as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            return None
+        else:
+            module.fail_json_aws(e)
+
+
+def has_component_params_changed(params, current_component_params):
+    """
+    Compare components (sans tags). If there is a difference, return True immediately else return False
+
+    :param params: the component params
+    :param current_component_params: the current state of the component params
+    :return: True if any parameter is mismatched else False
+    """
+
+    simple_comparison_keys = ['change_description', 'data',
+                              'description', 'kms_key_id', 'platform', 'semantic_version', 'uri']
+    for k in simple_comparison_keys:
+        if k in params and params[k] != current_component_params[k]:
+            return True
+
+    return False
+
+
+def have_component_tags_changed(params, current_component_params):
+    return 'tags' in params and params['tags'] != current_component_params['tags']
+
+
+def update_component_tags(connection, arn, params, current_component_params):
+    if len(params['tags']) == 0:
+        existing_tag_keys = list(current_component_params['tags'])
+        connection.untag_resource(
+            resourceArn=arn, tagKeys=existing_tag_keys)
+    else:
+        connection.tag_resource(resourceArn=arn, tags=params['tags'])
+
+
+def main():
+
+    argument_spec = (
+        dict(
+            change_description=dict(type='str'),
+            description=dict(type='str'),
+            data=dict(type='str'),
+            kms_key_id=dict(type='str'),
+            name=dict(required=True, type='str'),
+            platform=dict(type='str'),
+            semantic_version=dict(required=True, type='str'),
+            state=dict(required=True, choices=['present', 'absent'], type='str'),
+            tags=dict(type='dict'),
+            uri=dict(type='str')
+        )
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=[
+            ('state', 'present', ['platform'])
+        ]
+    )
+
+    connection = module.client('imagebuilder')
+
+    region = get_aws_connection_info(module, boto3=True)[0]
```

Comment:
rather than importing get_aws_connection_info we now have "module.region" available to us

```suggestion
    region = module.region
```